commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/ComplexTest.java

```diff
@@ -1419,6 +1419,9 @@ void testLog10() {
             final Complex z = Complex.ofCartesian(rng.nextDouble() * 2, rng.nextDouble() * 2);
             final Complex lnz = z.log();
             final Complex log10z = z.log10();
+            final ComplexNumber log10z2 = ComplexFunctions.log10(z.getReal(), z.getImaginary(), ComplexNumber::new);
```

Comment:
There is no test for ComplexFunctions.log.

In this case it may be convenient to add a method to TestUtils to assert two operations are the same, then return the result. You can then do further assertions on the result.

```Java
    static Complex assertSame(Complex c,
                              UnaryOperator<Complex> operation1,
                              ComplexUnaryOperator<ComplexNumber> operation2,
                              String name) {
        final Complex z = operation1.apply(c);
        // Test operation2 produces the exact same result
        operation2.apply(c.real(), c.imag(), (x, y) -> {
            Assertions.assertEquals(z.real(), x, () -> name + " real");
            Assertions.assertEquals(z.imag(), y, () -> name + " imaginary");
            return null;
        });
        return z;
    }

Complex lnz = TestUtils.assertSame(z, Complex::log, ComplexFunctions::log, "log");
Complex ln10z = TestUtils.assertSame(z, Complex::log10, ComplexFunctions::log10, "log10");
// Test lnz ...
```